libraries/chain/controller.cpp

```diff
@@ -531,22 +531,137 @@ struct controller_impl {
        // TODO: Removed by CyberWay
    }
 
-   void add_to_snapshot( const snapshot_writer_ptr& snapshot ) const {
-      // TODO: Removed by CyberWay
+    void add_to_snapshot( const snapshot_writer_ptr& snapshot ) const {
+        std::vector<cyberway::chaindb::abi_info> abis = dump_accounts(snapshot);
 
-      authorization.add_to_snapshot(snapshot);
-      resource_limits.add_to_snapshot(snapshot);
-   }
+        for (const auto& abi : abis) {
+           dump_contract(snapshot, abi);
+        }
+    }
 
-   void read_from_snapshot( const snapshot_reader_ptr& snapshot ) {
-      // TODO: Removed by CyberWay
+    std::vector<cyberway::chaindb::abi_info> dump_accounts(const snapshot_writer_ptr& snapshot) const {
+        std::vector<cyberway::chaindb::abi_info> abis;
+        snapshot->write_section("account_table", [&] (auto& section) {
```

Comment:
This section may be overwritten by `table` table from `account` contract.

Reply:
at line 557:
 if (cyberway::chaindb::is_system_code(abi.code()) && table.first == account_table::table_name()) 
this case is filtered out